### PR TITLE
Add 'Jump To Entry Point' context menu item

### PR DESCRIPTION
**Pull request by 1337vida**
This adds an function to the context menu item which automatically finds the entry point and jumps to it.

### CODE REVIEW

File: ILSpy/TreeNodes/AssemblyTreeNode.cs [hunk 1]
```diff
@@ -687,4 +687,35 @@ public void Execute(TextViewContext context)
...
+			}
+		}
+
+		public MethodDefinition GetEntryPoint(TextViewContext context)
+		{
+			if (context.SelectedTreeNodes == null || context.SelectedTreeNodes.Length != 1)
+				return null;
+
+			AssemblyTreeNode asmNode = context.SelectedTreeNodes[0] as AssemblyTreeNode;
+			return asmNode?.LoadedAssembly?.AssemblyDefinition?.EntryPoint;
```
Comment (siegfriedpammer) [inline ILSpy/TreeNodes/AssemblyTreeNode.cs:718]: This code will not work with the latest version of ILSpy. There is no `AssemblyDefinition` property on `LoadedAssembly`.

File: ILSpy/TreeNodes/AssemblyTreeNode.cs [hunk 2]
```diff
@@ -687,4 +687,35 @@ public void Execute(TextViewContext context)
 		}
 	}
 
+	[ExportContextMenuEntry(Header = nameof(Resources._JumpToEntryPoint))]
+	sealed class JumpToEntryPoint : IContextMenuEntry
+	{
+		public bool IsVisible(TextViewContext context)
+		{
+			return GetEntryPoint() != null;
```
Comment (siegfriedpammer) [inline ILSpy/TreeNodes/AssemblyTreeNode.cs:695]: The method is defined as taking one parameter.

File: ILSpy/TreeNodes/AssemblyTreeNode.cs [hunk 3]
```diff
@@ -687,4 +687,35 @@ public void Execute(TextViewContext context)
...
+		}
+
+		public bool IsEnabled(TextViewContext context) 
+		{
+			return true;
+		}
+
+		public void Execute(TextViewContext context)
+		{
+			MethodDefinition entryPoint = GetEntryPoint();
```
Comment (siegfriedpammer) [inline ILSpy/TreeNodes/AssemblyTreeNode.cs:705]: Here the `context ` parameter is missing as well...
It seems like you did not try to run this code before creating a pull request...

**Comment by siegfriedpammer** [comment]
Honestly I am not sure, if this feature is really necessary... It takes two clicks to use this feature: Right click + Left click on the menu item.
The same is necessary to click on the Assembly node, then move to the decompiler view and click on the entry point link:

image

**Comment by 1337vida** [comment]
Yes, but i think its an nice addition to have. I think its more intuitive to right click on assembly and click "Jump to Entry Point". Sorry for the broken code, I will fix it when I'm home.

**Comment by siegfriedpammer** [comment]
Are you going to open another pull request?

**Pull request closed without merging**